guides/plugins/apps/starter/product-translator.md

```diff
@@ -0,0 +1,398 @@
+    <meta>
+    <!-- ... -->
+    </meta>
+    <setup>
+        <registrationUrl>http://127.0.0.1:8000/register</registrationUrl>
+        <secret>01f17b06402f0a24e6d2b084a6d18a87</secret>
+    </setup>
+</manifest>
+```
+{% endcode %}
+
+The `<registraionUrl>` is already implementend by the app template, and is always `/register`, unless you modify `src/Controller/RegistrationController.php`.
+The `<secret>` element is only present in development versions of the app. In production, the extension store will provide the secret to authenticate your app buyers.
+
+Because this app will need to read product descriptions and translate them the it needs permissions to do so:
+
+{% code title="release/manifest.xml" %}
+``` xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/master/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+    <!-- ... -->
+    </meta>
+    <setup>
+    <!-- ... -->
+    </setup>
+    <permissions>
+        <read>product</read>
+        <read>product_translation</read>
+        <read>language</read>
+        <update>product_translation</update>
+        <create>product_translation</create>
+    </permissions>
+</manifest>
+```
+{% endcode %}
+
+Finally, your app needs to be notified every time a product description is modified.
+The app system provides webhooks to subscribe your app server to any changes in the data
+in its shops:
+
+{% code title="release/manifest.xml" %}
+``` xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/master/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+    <!-- ... -->
+    </meta>
+    <setup>
+    <!-- ... -->
+    </setup>
+    <permissions>
+    <!-- ... -->
+    </permissions>
+    <webhooks>
+        <webhook name="deleted" url="http://127.0.0.1:8000/registration/remove" event="app.deleted"></webhook>
+        <webhook name="product-update" url="http://127.0.0.1:8000/product/written" event="product.written"></webhook>
+    </webhooks>
+</manifest>
+```
+{% endcode %}
+
+These two webhooks provide a way for shops to notify your app server about events that occurred.
+The `src/Controller/RegistrationController.php`  controller in the app template already provides the `deleted` webhook. It notifies the server that a shop has deleted the app.
+
+The `product-update` webhook is the path on which your app server will be notified about any product updates in the stores, like changing the description.
+
+This webhook needs a custom controller, which will be the next part of this guide.
+
+## Implementing the translation
```

Comment:
Please improve readability by adding 3rd-level sub-headings where applicable

Reply:
Also I'm thinking "Implementing the translation" is not the correct heading, as the actual translation is not a part of the guide. Maybe more like "Implementing the webhook listener" (not happy with that one either, but it's closer to what we're describing).